api/utils/prompt/stdin.go

```diff
@@ -69,3 +74,36 @@ func NotifyExit() {
 	}
 	stdinMU.Unlock()
 }
+
+// EnableStdinTerminalFallback attempts to use a fallback like /dev/tty when stdin is
+// not the terminal. This can be useful when tsh is called by another terminal
+// tool (e.g. git).
+func EnableStdinTerminalFallback() {
+	stdinMU.Lock()
+	defer stdinMU.Unlock()
+	stdinTerminalFallback = true
+}
+
+func maybeUseStdinTerminalFallbackLocked(original *ContextReader) *ContextReader {
+	if !stdinTerminalFallback || runtime.GOOS == "windows" || original.IsTerminal() {
+		return original
+	}
+
+	// File /dev/tty is the controlling tty of the current terminal. Not
+	// available on Windows.
+	// https://tldp.org/HOWTO/Text-Terminal-HOWTO-7.html
+	devTTY, err := os.Open("/dev/tty")
```

Comment:
Have you considered using a library for this, for example: `https://github.com/mattn/go-tty`? This particular one is fairly small and as a bonus it supports Windows.

Reply:
that looks promising, let me give that a try. thanks!